prov/rxm/src/rxm_conn.c

```diff
@@ -1361,19 +1403,161 @@ static void *rxm_conn_progress(void *arg)
 		entry->rd = rxm_eq_sread(rxm_ep, RXM_CM_ENTRY_SZ, entry);
 		if (entry->rd < 0 && entry->rd != -FI_ECONNREFUSED)
 			goto exit;
-
-		if (entry->event == FI_NOTIFY &&
-		    (enum rxm_cmap_signal)((struct fi_eq_entry *)
-					   &entry->cm_entry)->data == RXM_CMAP_EXIT) {
-			FI_DBG(&rxm_prov, FI_LOG_EP_CTRL,
-			       "Closing CM thread\n");
+		if (rxm_conn_eq_event(rxm_ep, entry))
 			goto exit;
+		memset(entry, 0, RXM_MSG_EQ_ENTRY_SZ);
+	}
+exit:
```

Comment:
nit: We can remove this label and replace the above goto calls with break.  Can likely use alloca() to allocate entry instead of calloc(), but means moving the memset() to the top of the loop.

Reply:
I added a separate commit that applies these recommendations to the original CM thread auto progress function. Thanks!